lesson16/index.js

```diff
@@ -0,0 +1,158 @@
+      const img = document.createElement("img");
+      img.src = content.image.img;
+      primaryImageWrapper.appendChild(img);
+
+      renderNewsList(content, lists);
+      const tabItem = document.querySelector(
+        ".news_Tabs .tab_item:nth-child(1)"
+      );
+      tabItem.classList.add("is-active");
+    }
+  });
+
+  // タブクリック時の処理
+  newsTabs.addEventListener("click", (event) => {
+    // event.targetにis-activeをつける
+    // event.targetのidを取得し、それに紐づく記事と画像を取得
+    const target = event.target;
+    const tabsItems = Array.from(newsTabs.querySelectorAll("li"));
+
+    // タブのindexを取得し、スタイル変更
+    tabsItems.forEach((tab) => {
+      tab.addEventListener("click", (e) => {
+        const activeIndex = tabsItems.filter((tab) => {
+          return tab === e.target;
+        });
+        const inactiveIndex = tabsItems.filter((tab) => {
+          return tab !== e.target;
+        });
+
+        activeIndex[0].classList.add("is-active");
+        inactiveIndex.forEach((element) => {
+          element.classList.remove("is-active");
+        });
+      });
+    });
+
+    if (target.textContent === "news") {
+      console.log("news");
+
+      const imgWrapper = document.querySelector(".news_Img");
+      imgWrapper.innerHTML = "";
+      const img = document.createElement("img");
+      img.src = allData[0].image.img;
+      img.alt = allData[0].image.alt;
+      imgWrapper.appendChild(img);
+
+      renderNewsList(allData[0], lists);
+    } else if (event.target.textContent === "economy") {
+      console.log("economy");
+
+      const imgWrapper = document.querySelector(".news_Img");
+      imgWrapper.innerHTML = "";
+      const img = document.createElement("img");
+      img.src = allData[1].image.img;
+      img.alt = allData[1].image.alt;
+      imgWrapper.appendChild(img);
+
+      renderNewsList(allData[1], lists);
+    } else if (event.target.textContent === "sports") {
+      console.log("sports");
+
+      const imgWrapper = document.querySelector(".news_Img");
+      imgWrapper.innerHTML = "";
+      const img = document.createElement("img");
+      img.src = allData[2].image.img;
+      img.alt = allData[2].image.alt;
+      imgWrapper.appendChild(img);
+
+      renderNewsList(allData[2], lists);
+    }
+  });
+};
+
+const renderNewsList = (content, lists) => {
+  const newsList = document.querySelector(".news_Lists");
+  newsList.innerHTML = "";
+  Object.keys(content.article).forEach((key) => {
+    const li = document.createElement("li");
+    const a = document.createElement("a");
+    const comment = document.createElement("span");
+    a.href = content.article[key].url;
+    a.textContent = content.article[key].title;
+    const newMark = document.createElement("span");
+    newMark.classList.add("newMark");
+    newMark.innerText = "new!";
+    const dateDiff = getDateDiff(content.article[key].publishedDate);
+    if (dateDiff < 3) {
+      a.appendChild(newMark);
+    }
```

Comment:
IMO
If the case is within 3 days, it would be more appropriate to generate and add the element.
```suggestion
    
    const dateDiff = getDateDiff(content.article[key].publishedDate);
    if (dateDiff < 3) {
    const newMark = document.createElement("span");
    newMark.classList.add("newMark");
    newMark.innerText = "new!";
     a.appendChild(newMark);
    }
```

Reply:
Sure, I thought it would be better to generate it only when needed, so I fixed it.